pkg/api/metrics_explore.go

```diff
@@ -0,0 +1,53 @@
+package api
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+func ExploreMetrics(c *middleware.Context) Response {
+	var result []string
+	filePath := setting.GraphiteMetrics
+	f, err := os.Open(filePath)
```

Comment:
Would it make sense to cache the file object instead of opening it whenever a new request is issued? I also know little to nothing about Go so its very possible that its natural magic is taking care of this for us.